File: src/yt/item5/bean/Brand.java
package yt.item5.bean;

import java.io.Serializable;
import java.util.Set;

public class Brand implements Serializable, EntityInterface {

	private static final long serialVersionUID = 1L;

	private int brandId;

	private String brandName;

	private String website;

	private String country;

	private Set<Shoes> shoesGroup;

	public Brand() {
	}

	public Brand(String brandName) {
		this.brandName = brandName;
	}

	public Brand(int brandId) {
		this.brandId = brandId;
	}

	public Brand(int brandId, String brandName, String website, String country) {
		this.brandId = brandId;
		this.brandName = brandName;
		this.website = website;
		this.country = country;
	}

	public int getBrandId() {
		return brandId;
	}

	public Brand setBrandId(int id) {
		this.brandId = id;
		return this;
	}

	public String getBrandName() {
		return brandName;
	}

	public Brand setBrandName(String brandName) {
		this.brandName = brandName;
		return this;
	}

	public String getWebsite() {
		return website;
	}

	public Brand setWebsite(String website) {
		this.website = website;
		return this;
	}

	public String getCountry() {
		return country;
	}

	public Brand setCountry(String country) {
		this.country = country;
		return this;
	}

	@Override
	public String toString() {
		return brandId + " " + brandName + " " + website + " " + country;

	}

	public Set<Shoes> getShoesGroup() {
		return shoesGroup;
	}

	public void setShoesGroup(Set<Shoes> shoesGroup) {
		this.shoesGroup = shoesGroup;
	}

	public void addShoes(Shoes shoes) {
		shoesGroup.add(shoes);
	}

	public void removeShoes(Shoes shoes) {
		shoesGroup.remove(shoes);
	}

	@Override
	public int getId() {
		return this.brandId;
	}

}

File: WebContent/js/jsModel/Brand.js
function Brand() {
	var brandId, brandName, website, country;
	var shoesGroup = new Array();

	this.setBrandId = function(param) {
		brandId = param;
		return this;
	}
	this.setBrandName = function(param) {
		brandName = param;
		return this;
	}
	this.setWebsite = function(param) {
		website = param;
		return this;
	}
	this.setCountry = function(param) {
		country = param;
		return this;
	}
	this.setAttributeByObj = function(obj) {
		brandId = obj['brandId'];
		brandName = obj['brandName'];
		website = obj['website'];
		country = obj['country'];
		return this;
	}
	this.getAttributes = function() {
		return {
			'brandId' : brandId,
			'brandName' : brandName,
			'website' : website,
			'country' : country
		};
	}
	this.getBrandId = function() {
		return brandId;
	}
	this.getBrandName = function() {
		return brandName;
	}
	this.getWebsite = function() {
		return website;
	}
	this.getCountry = function() {
		return country;
	}
};

Brand.prototype = (function() {

	return {
		colAttribute : [ 'brandId',
			'brandName',
			'website',
			'country' ],
		entityServlet : 'BrandTableController',
		inputColAttributeMap : {
			brandId : {
				colTag : 'input',
				colType : 'text',
				otherAttribute : 'readOnly'
			},
			brandName : {
				colTag : 'input',
				colType : 'text',
				otherAttribute : 'requierd'
			},
			website : {
				colTag : 'input',
				colType : 'text'
			},
			country : {
				colTag : 'select'
			}
		}

	}
}());